### PR TITLE
AppAuthMigration doc update

### PR DESCRIPTION
**Pull request by nonik0**
Updating AppAuth migration documentation with new guidance for newer scenarios supported in Azure.Identity.

### CODE REVIEW

File: api/overview/azure/latest/app-auth-migration.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ If you are starting development of a new application, it is strongly recommended
 
 |Auth Provider|AppAuthentication<br>Connection String|Azure.Identity<br>TokenCredential|
 |---------|---------|---------|
-|Default - environment-based | Default - no connection string used | new DefaultAzureCredential()*|
+|Default / environment-based | Default - no connection string used | new DefaultAzureCredential()*|
```
Comment (anaismiller) [inline api/overview/azure/latest/app-auth-migration.md:32]:
```suggestion
|(Default) environment-based | Default - no connection string used | new DefaultAzureCredential()*|
```

File: api/overview/azure/latest/app-auth-migration.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ If you are starting development of a new application, it is strongly recommended
 
 > [!NOTE]
 > `*` Authentication providers and order is different between [AzureServiceTokenProvider](https://github.com/Azure/azure-sdk-for-net/blob/7d23a9d912da40baeebee1125eb5ebefa78449a2/sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs#L104) and [DefaultAzureCredential](https://docs.microsoft.com/dotnet/api/overview/azure/identity-readme#defaultazurecredential)<br>
-> `**` Need to set [environment variables](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#environment-variables)
+> `**` Need to set [environment variables](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme#environment-variables)
```
Comment (anaismiller) [inline api/overview/azure/latest/app-auth-migration.md:45]:
```suggestion
> `**` Need to set [environment variables](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme#environment-variables)
```

File: api/overview/azure/latest/app-auth-migration.md [hunk 3]
```diff
@@ -51,35 +51,38 @@ While Azure.Identity supports most authentication scenarios and providers that A
 -   System.Data.SqlClient.SqlAuthenticationProvider implementation
     (SqlAppAuthenticationProvider)
 
+    -   For Microsoft.Data.SqlClient, see [Active Directory Default authentication](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-default-authentication). This authentication mode provides similar functionality where DefaultAzureCredential is used to obtain access token for authentication to SQL instance
```
Comment (anaismiller) [inline api/overview/azure/latest/app-auth-migration.md:54]:
```suggestion
    -   For Microsoft.Data.SqlClient, see [Active Directory Default authentication](https://learn.microsoft.com/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-default-authentication). This authentication mode provides similar functionality where DefaultAzureCredential is used to obtain access token for authentication to SQL instance
```

File: api/overview/azure/latest/app-auth-migration.md [hunk 4]
```diff
@@ -51,35 +51,38 @@ While Azure.Identity supports most authentication scenarios and providers that A
...
+
 -   Directly use certificates in cert store as client credential (using subject name or thumbprint identifier)
 
 -   Directly use certificates in Key Vault as client credential (using Key Vault certificate secret identifier)
 
 -   Change authentication provider with environment config (i.e. connection strings in AppAuthentication)
 
-    -   Limited support in Azure.Identity with EnvironmentCredential
+    -   Limited support in Azure.Identity with DefaultAzureCredential and EnvironmentCredential, see [environment variables](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme#environment-variables)
```
Comment (anaismiller) [inline api/overview/azure/latest/app-auth-migration.md:62]:
```suggestion
    -   Limited support in Azure.Identity with DefaultAzureCredential and EnvironmentCredential, see [environment variables](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme#environment-variables)
```

File: api/overview/azure/latest/app-auth-migration.md [hunk 5]
```diff
@@ -51,35 +51,38 @@ While Azure.Identity supports most authentication scenarios and providers that A
...
 
 -   Determine auth provider and identity used for environment-based authentication (i.e. AzureServiceTokenProvider.PrincipalUsed property)
 
+    - Can determine auth provider used with DefaultAzureCredential in Azure.Identity using [AzureEventSourceListener](https://learn.microsoft.com/en-us/dotnet/api/azure.core.diagnostics.azureeventsourcelistener)
```
Comment (anaismiller) [inline api/overview/azure/latest/app-auth-migration.md:66]:
```suggestion
    - Can determine auth provider used with DefaultAzureCredential in Azure.Identity using [AzureEventSourceListener](https://learn.microsoft.com/dotnet/api/azure.core.diagnostics.azureeventsourcelistener)
```

**Comment by anaismiller** [comment]
@jsquire to review as well and help with merge, thank you.

File: api/overview/azure/latest/app-auth-migration.md (hunk 3 above)
Comment (jsquire) [inline api/overview/azure/latest/app-auth-migration.md:54]:
```suggestion
    -   For Microsoft.Data.SqlClient, see [Active Directory Default authentication](https://learn.microsoft.com/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-default-authentication). This authentication mode provides similar functionality where DefaultAzureCredential is used to obtain access token for authentication to SQL instances.
```

Comment (jsquire) [reply api/overview/azure/latest/app-auth-migration.md:54]: Anais caught the culture slug already, but it seemed weird to leave it out of the suggestion...

**Comment by jsquire** [comment]
When things are ready to merge, please mention me and I'd be happy to help with the pushing of buttons.

File: api/overview/azure/latest/app-auth-migration.md [hunk 6]
```diff
@@ -134,4 +130,5 @@ var accessToken = await tokenCredential.GetTokenAsync(
 );
 ```
 
-Note: More information on the `.default` scope can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#the-default-scope).
+> [!NOTE]
+More information on the `.default` scope can be found [here](https://learn.microsoft.com/azure/active-directory/develop/scopes-oidc#the-default-scope).
```
Comment (anaismiller) [inline api/overview/azure/latest/app-auth-migration.md:134]:
```suggestion
> More information on the `.default` scope can be found [here](https://learn.microsoft.com/azure/active-directory/develop/scopes-oidc#the-default-scope).
```

**Review by anaismiller — APPROVED** [review]

**Comment by nonik0** [comment]
@jsquire Should be all good to go now, thanks!

**Review by jsquire — APPROVED** [review]

**Pull request merged**